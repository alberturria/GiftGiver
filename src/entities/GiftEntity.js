export default class GiftEntity{
    constructor(id, name, description) {
        this.id = id;
        this.name = name;
        this.description = description;
    }
}