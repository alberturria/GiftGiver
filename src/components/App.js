import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import GiftEntity from '../entities/GiftEntity';
import { maxNumber } from '../utils';
import Gift from './Gift';
import './index.css';

export default class App extends Component {
    constructor() {
        super();

        this.state = { gifts: [] }
    }

    addGift = () => {
        const { gifts } = this.state;
        const existingIds = gifts.map( gift => gift.id);
        const maxId = maxNumber(existingIds);
        const newGift = new GiftEntity(maxId + 1)
        gifts.push(newGift);

        this.setState({ gifts });
    }

    _renderGifts = () => {
        const { gifts } = this.state;

        return gifts.map(gift => <Gift key={gift.id} gift={new GiftEntity(gift.id)} removeGiftCallback={this.removeGift} />);
    }

    removeGift = (giftId) => {
        const { gifts } = this.state;
        const newGifts = gifts.filter(gift => gift.id !== giftId);

        this.setState( {gifts: newGifts });
    }

    render() {
        return (
            <div>
                <h2>Gift Giver</h2>
                <div className='gifts-list'>
                    {this._renderGifts()}
                </div>
                <Button className='button-add-gift' onClick={this.addGift}>Add Gift</Button>
            </div>
        )
    }
}