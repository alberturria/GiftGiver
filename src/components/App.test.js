import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import GiftEntity from '../entities/GiftEntity';

const app = shallow(<App />);


it('should render the react root node', () => {
    expect(app).toMatchSnapshot();
})

it('Should initialize the `state` as am empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
})

it('Should add a new Gift to the `state` when clicking on the `add gift` button', () => {
    const expectedGiftEntity = new GiftEntity(0);
    const addGiftButton = app.find('.button-add-gift');
    addGiftButton.simulate('click');
    const stateGifts = app.state().gifts;
    
    expect(stateGifts).toEqual([expectedGiftEntity]);
})

it('Should add a new Gift to the gifts list when clickin on the add Gift Button', () => {
    const expectedNumberOfGifts = 1;
    const addGiftButton = app.find('.button-add-gift');
    addGiftButton.simulate('click');
    const giftList = app.find('.gifts-list');
    const numberOfGifts = giftList.children.length;
    
    expect(numberOfGifts).toEqual(expectedNumberOfGifts);
})