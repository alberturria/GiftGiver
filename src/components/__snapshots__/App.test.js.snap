// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the react root node 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <div
    className="gifts-list"
  />
  <Button
    active={false}
    className="button-add-gift"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Add Gift
  </Button>
</div>
`;
